### PR TITLE
integration: TestCacheControl and TestHSTS close the ResponseBody

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/apiserver/apiserver_test.go [hunk 1]
```diff
@@ -246,6 +246,7 @@ func TestCacheControl(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
+			resp.Body.Close()
```
Comment (wojtek-t) [inline test/integration/apiserver/apiserver_test.go:249]: Can we do that in defer?

Comment (p0lyn0mial) [reply test/integration/apiserver/apiserver_test.go:249]: we can but why :) ?

Comment (p0lyn0mial) [reply test/integration/apiserver/apiserver_test.go]: done.

**Pull request merged**